modules/agentless-s3-bucket/main.tf

```diff
@@ -0,0 +1,73 @@
+locals {
+  dd_tags = {
+    Datadog                 = "true"
+    DatadogAgentlessScanner = "true"
+  }
+}
+
+data "aws_region" "current" {}
+
+resource "aws_s3_bucket" "bucket" {
+  bucket_prefix = "datadog-agentless-scanning-"
+  tags          = merge(var.tags, local.dd_tags)
+}
+
+resource "aws_s3_bucket_public_access_block" "bucket_access_block" {
+  bucket = aws_s3_bucket.bucket.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "bucket_lifecycle" {
+  bucket = aws_s3_bucket.bucket.id
+  rule {
+    status = "Enabled"
+    id     = "expire_all_files"
+    expiration {
+      days = 2
+    }
+  }
+}
+
+data "aws_iam_policy_document" "bucket_access_policy_document" {
+  # TODO: add statement to deny access to everyone except the scanner roles
+  statement {
+    sid = "DatadogAgentlessScannerBucketPolicy"
+
+    principals {
+      type = "AWS"
+      identifiers = [
+        var.iam_delegate_role_arn,
+        var.iam_rds_assume_role_arn,
+      ]
+    }
+
+    actions = [
```

Comment:
I would avoid wildcards in these actions and instead list them all. You can use https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html to find out all the actions available

Reply:
I just followed [aws documentation](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_ExportSnapshot.html#USER_ExportSnapshot.Setup) under the section "To give DB snapshot tasks access to Amazon S3".
But following your advice, I split the policies for the service role and the delegate role and I narrowed down the permissions for the delegate role.

Reply:
Yeah the AWS documentation and managed policies are sometimes overprivileged or suboptimal 😬 Thanks!